### DISCUSSION TITLE
Dataset research + resources

### DISCUSSION THREAD

**Discussion by mrdbourke** [mrdbourke/nutrify#23] [General]
Starting this to take note of different food image datasets out there.

The goal will be to eventually have the largest food image dataset on the internet.

Post comments of anything you find and I'll collate them here.

## Datasets online

We'll need *lots* of food images, so the more we can find open source (before the app is built and we start collecting custom images), the better.

* 800k food images from Instagram - https://homepages.inf.ed.ac.uk/thospeda/downloads.html 
* PASS, 1.4M images like ImageNet but without people (this is good because Nutrify will train on images of food but not people) - https://www.robots.ox.ac.uk/~vgg/research/pass/
* More images of food via Papers With Code Dataset page - https://paperswithcode.com/datasets?q=food&v=lst&o=match 

From @ashikshafi08 
-  [FoodX-251 Dataset | Papers With Code](https://paperswithcode.com/dataset/foodx-251)
- [Nutrition 5k dataset ](https://storage.cloud.google.com/nutrition5k_dataset/nutrition5k_dataset.tar.gz)
- [simplified-recipes-1M Dataset - Dominik Schmidt](https://dominikschmidt.xyz/simplified-recipes-1M/)
- [Epirecipes ](https://www.kaggle.com/hugodarwood/epirecipes)
- [Recipe Ingredients Dataset | Kaggle](https://www.kaggle.com/kaggle/recipe-ingredients-dataset)
-  [Recipe1M+: A Dataset for Learning Cross-Modal Embeddings for Cooking Recipes and Food Images - MIT](http://pic2recipe.csail.mit.edu/)

## Nutrition APIs

When Nutrify classifies an image, we'll want to display the food's appropriate nutrition information, below are a list of potential nutrition APIs.

* Top 12 nutrition APIs - https://rapidapi.com/collection/nutrition
* US Department of agriculture API - https://fdc.nal.usda.gov/api-guide.html
* MyNetDiary food API (cost) - https://www.mynetdiary.com/food-database.html

## Labeling and data exploration

- Can use FitfyOne to explore data, download popular datasets and find mistakes - https://voxel51.com/docs/fiftyone/tutorials/open_images.html

**Comment by ashikshafi08** [comment]
So far I have found these resources and this will be updated as I find new things. 


## Papers 
- [Inverse Cooking: Recipe Generation from Food Images | Papers With Code](https://paperswithcode.com/paper/inverse-cooking-recipe-generation-from-food)
- [Nutrition5k: Towards Automatic Nutritional Understanding of Generic Food | Papers With Code](https://paperswithcode.com/paper/nutrition5k-towards-automatic-nutritional)
- [DeepFood: Deep Learning-Based Food Image Recognition for Computer-Aided Dietary Assessment](https://paperswithcode.com/paper/deepfood-deep-learning-based-food-image)
- [Food Ingredients Recognition through Multi-label Learning](https://paperswithcode.com/paper/food-ingredients-recognition-through-multi)
- [Deep Learning-Based Food Calorie Estimation Method in Dietary Assessment | Papers With Code](https://paperswithcode.com/paper/deep-learning-based-food-calorie-estimation)
- [RecipeGPT: Generative Pre-training Based Cooking Recipe Generation and Evaluation System | Papers With Code](https://paperswithcode.com/paper/recipegpt-generative-pre-training-based)
- [Now You’re Cooking! Benchmark (Recipe Generation) | Papers With Code](https://paperswithcode.com/sota/recipe-generation-on-now-youre-cooking)
- [Food.com](https://paperswithcode.com/sota/recipe-generation-on-foodcom)
- [RecipeNLG: A Cooking Recipes Dataset for Semi-Structured Text
Generation](https://aclanthology.org/2020.inlg-1.4.pdf)

## Datasets 
-  [FoodX-251 Dataset | Papers With Code](https://paperswithcode.com/dataset/foodx-251)
- [Nutrition 5k dataset ](https://storage.cloud.google.com/nutrition5k_dataset/nutrition5k_dataset.tar.gz)
- [simplified-recipes-1M Dataset - Dominik Schmidt](https://dominikschmidt.xyz/simplified-recipes-1M/)
- [Epirecipes ](https://www.kaggle.com/hugodarwood/epirecipes)
- [Recipe Ingredients Dataset | Kaggle](https://www.kaggle.com/kaggle/recipe-ingredients-dataset)
-  [Recipe1M+: A Dataset for Learning Cross-Modal Embeddings for Cooking Recipes and Food Images - MIT](http://pic2recipe.csail.mit.edu/)

**Reply by mrdbourke** [reply]
Thank you thank you thank you!

**Comment by mrdbourke** [comment]
Good amount of food data with image instance segmentation - https://xiongweiwu.github.io/foodseg103.html
